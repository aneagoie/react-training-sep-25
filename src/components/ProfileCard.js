import React from 'react';
import PropTypes from 'prop-types';
import { Heading } from './Heading';
import { Text } from './Text';
import { Avatar } from './Avatar';

export const ProfileCard = ({ uid, name, email, className, ...props }) => (
  <article
    className={`pa3 br2 tc bg-near-white grow pointer ${className}`}
    {...props}
  >
    <Avatar uid={uid} />
    <Heading level={2} size={4}>
      {name}
    </Heading>
    <Text>{email}</Text>
  </article>
);

ProfileCard.propTypes = {
  uid: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired,
};
