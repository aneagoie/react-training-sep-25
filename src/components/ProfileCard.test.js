import React from 'react';
import renderer from 'react-test-renderer';
import { ProfileCard } from './ProfileCard';

it('ProfileCard', () => {
  const tree = renderer
    .create(<ProfileCard uid={1} name="Alice X" email="alice@test.com" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
