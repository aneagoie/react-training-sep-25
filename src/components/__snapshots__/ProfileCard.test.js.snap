// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileCard 1`] = `
<article
  className="pa3 br2 tc bg-near-white grow pointer "
>
  <img
    className="w4 h4 br-100"
    src="https://robohash.org/1"
  />
  <h2
    className="sans-serif dark-gray f4"
  >
    Alice X
  </h2>
  <p
    className="sans-serif dark-gray"
  >
    alice@test.com
  </p>
</article>
`;
